java/vector/src/test/java/org/apache/arrow/vector/util/TestVectorAppender.java

```diff
@@ -0,0 +1,357 @@
+
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.RootAllocator;
+import org.apache.arrow.vector.BitVectorHelper;
+import org.apache.arrow.vector.IntVector;
+import org.apache.arrow.vector.VarCharVector;
+import org.apache.arrow.vector.complex.FixedSizeListVector;
+import org.apache.arrow.vector.complex.ListVector;
+import org.apache.arrow.vector.complex.StructVector;
+import org.apache.arrow.vector.complex.UnionVector;
+import org.apache.arrow.vector.holders.NullableBigIntHolder;
+import org.apache.arrow.vector.holders.NullableFloat4Holder;
+import org.apache.arrow.vector.holders.NullableIntHolder;
+import org.apache.arrow.vector.types.Types;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.FieldType;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Test cases for {@link VectorAppender}.
+ */
+public class TestVectorAppender {
+
+  private BufferAllocator allocator;
+
+  @Before
+  public void prepare() {
+    allocator = new RootAllocator(1024 * 1024);
+  }
+
+  @After
+  public void shutdown() {
+    allocator.close();
+  }
+
+  @Test
+  public void testAppendFixedWidthVector() {
+    final int length1 = 10;
+    final int length2 = 5;
+    try (IntVector target = new IntVector("", allocator);
+         IntVector delta = new IntVector("", allocator)) {
+
+      target.allocateNew(length1);
+      delta.allocateNew(length2);
+
+      for (int i = 0; i < length1; i++) {
+        target.set(i, i);
+      }
+      for (int i = 0; i < length2; i++) {
+        delta.set(i, i + length1);
+      }
+      target.setValueCount(length1);
+      delta.setValueCount(length2);
+
+      VectorAppender appender = new VectorAppender(target);
+      delta.accept(appender, null);
+
+      assertEquals(length1 + length2, target.getValueCount());
+      for (int i = 0; i < target.getValueCount(); i++) {
+        assertEquals(i, target.get(i));
+      }
+    }
+  }
+
+  @Test
+  public void testAppendVariableWidthVector() {
+    final int length1 = 10;
+    final int length2 = 5;
+    try (VarCharVector target = new VarCharVector("", allocator);
+         VarCharVector delta = new VarCharVector("", allocator)) {
+
+      target.allocateNew(5, length1);
+      delta.allocateNew(5, length2);
+
+      for (int i = 0; i < length1; i++) {
+        target.setSafe(i, ("a" + i).getBytes());
+      }
+      for (int i = 0; i < length2; i++) {
+        delta.setSafe(i, ("a" + (i + length1)).getBytes());
+      }
+      target.setValueCount(length1);
+      delta.setValueCount(length2);
+
+      VectorAppender appender = new VectorAppender(target);
+      delta.accept(appender, null);
+
+      assertEquals(length1 + length2, target.getValueCount());
+      for (int i = 0; i < target.getValueCount(); i++) {
+        assertEquals("a" + i, new String(target.get(i)));
+      }
+    }
+  }
+
+  private ListVector createListVector(int start, int end, int step) {
```

Comment:
see comment below on taking literal values instead of created values in loops.  having a seperate method here make it harded to understand the actual test.

Reply:
I have revised the test cases accordingly. Thanks.